ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/validation/SignedBlsToExecutionChangeValidator.java

```diff
@@ -13,32 +13,123 @@
 
 package tech.pegasys.teku.statetransition.validation;
 
+import static tech.pegasys.teku.spec.config.Constants.VALID_VALIDATOR_SET_SIZE;
+import static tech.pegasys.teku.statetransition.validation.ValidationResultCode.IGNORE;
+
 import java.util.Optional;
+import java.util.Set;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.bls.BLSSignatureVerifier;
 import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.collections.LimitedSet;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.datastructures.operations.BlsToExecutionChange;
 import tech.pegasys.teku.spec.datastructures.operations.SignedBlsToExecutionChange;
 import tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState;
 import tech.pegasys.teku.spec.logic.common.operations.validation.OperationInvalidReason;
+import tech.pegasys.teku.storage.client.RecentChainData;
 
 public class SignedBlsToExecutionChangeValidator
     implements OperationValidator<SignedBlsToExecutionChange> {
 
-  // FIXME Implement validation logic (#6358)
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final Spec spec;
+  private final RecentChainData recentChainData;
+  private final Set<UInt64> seenBlsToExecutionChangeMessageFromValidators =
+      LimitedSet.createSynchronized(VALID_VALIDATOR_SET_SIZE);
+
+  public SignedBlsToExecutionChangeValidator(final Spec spec, RecentChainData recentChainData) {
+    this.spec = spec;
+    this.recentChainData = recentChainData;
+  }
 
   @Override
   public SafeFuture<InternalValidationResult> validateFully(
```

Comment:
`validateFully` is the method used for validating gossip messages. In this case, we are applying these rules: https://github.com/ethereum/consensus-specs/blob/dev/specs/capella/p2p-interface.md#bls_to_execution_change

p.s. this method will be renamed in a followup PR.